patterns/Behavioral/Strategy/php/strategy.php

```diff
@@ -0,0 +1,78 @@
+
+<?php
+//In this example we will implement the Strategy pattern in PHP for a string filtering system. 
+//We’ll create different strategies to filter out specific words or phrases from user input.
+
+
+//Define the strategy interface
+interface StringFilterStrategy {
+    public function filter($input);
+}
+
+// Concrete strategy classes
+class BlacklistFilter implements StringFilterStrategy {
+    private $blacklist;
+
+    public function __construct(array $blacklist) {
+        $this->blacklist = $blacklist;
+    }
+
+    public function filter($input) {
+        foreach ($this->blacklist as $word) {
+            $input = str_ireplace($word, '***', $input);
+        }
+        return $input;
+    }
+}
+
+class WhitelistFilter implements StringFilterStrategy {
+    private $whitelist;
+
+    public function __construct(array $whitelist) {
+        $this->whitelist = $whitelist;
+    }
+
+    public function filter($input) {
+        $filteredWords = [];
+        foreach (explode(' ', $input) as $word) {
+            if (in_array($word, $this->whitelist)) {
+                $filteredWords[] = $word;
+            }
+        }
+        return implode(' ', $filteredWords);
+    }
+}
+
+// Context class
+class StringFilterContext {
+    private $strategy;
+
+    public function setStrategy(StringFilterStrategy $strategy) {
+        $this->strategy = $strategy;
+    }
+
+    public function applyFilter($input) {
+        return $this->strategy->filter($input);
+    }
+}
```

Comment:
this is completely an acceptable way of writing this, but I really love using static method calls for this like this!

I would change it to

```PHP
class Filter {
  private static $strategy;
  
  public static function setStrategy(StringFilterStrategy $strategy) {
     self::$strategy = $strategy;
  }

  public static function applyFilter($input) {
      return self::$strategy->filter($input);
  }
}
```
by changing the name to filter, we're making it easier to call!